src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
This ternary expression may be difficult to understand. It would be better to simplify it by using a single `filter` method and incorporating the logic of the `withChildren` parameter within it. Additionally, the `find` method can be replaced with `some` as it is only necessary to check whether the women have children, not to find a specific one.

A possible solution here could be:
```
  const women = people.filter((person) => {
    const hasChildren = withChildren
      ? people.some((innerPerson) => innerPerson.mother === person.name)
      : true

    return person.sex === 'f' && hasChildren
  })
```